src/routes/consent.ts

```diff
@@ -1,60 +1,129 @@
 // Copyright © 2023 Ory Corp
 // SPDX-License-Identifier: Apache-2.0
-import { defaultConfig, RouteCreator, RouteRegistrator } from "../pkg"
+import { defaultConfig, logger, RouteCreator, RouteRegistrator } from "../pkg"
 import { register404Route } from "./404"
 import { oidcConformityMaybeFakeSession } from "./stub/oidc-cert"
-import {
-  AcceptOAuth2ConsentRequestSession,
-  IdentityApi,
-  OAuth2ConsentRequest,
-} from "@ory/client"
+import { AcceptOAuth2ConsentRequestSession } from "@ory/client"
 import { UserConsentCard } from "@ory/elements-markup"
 import bodyParser from "body-parser"
-import csrf from "csurf"
+import { doubleCsrf, DoubleCsrfCookieOptions } from "csrf-csrf"
+import { Request, Response, NextFunction } from "express"
 
-async function createOAuth2ConsentRequestSession(
-  grantScopes: string[],
-  consentRequest: OAuth2ConsentRequest,
-  identityApi: IdentityApi,
-): Promise<AcceptOAuth2ConsentRequestSession> {
-  // The session allows us to set session data for id and access tokens
+const cookieOptions: DoubleCsrfCookieOptions = {
+  sameSite: "lax",
+  signed: true,
+  // set secure cookies by default (recommended in production)
+  // can be disabled through DANGEROUSLY_DISABLE_SECURE_COOKIES=true env var
+  secure: true,
+  ...(process.env.DANGEROUSLY_DISABLE_SECURE_CSRF_COOKIES && {
+    secure: false,
+  }),
```

Comment:
This just checks, whether the env var is set, not whether it's set to "true", right?

Also, why not
```suggestion
  secure: process.env.DANGEROUSLY_DISABLE_SECURE_CSRF_COOKIES !== "true",
```
?

Reply:
I didn't want it to be ambiguous and rather made it as explicit as possible so that there would be no chance of a mess-up in future or a misunderstanding of what it does.